test/built-ins/Array/fromAsync/asyncitems-arraylike-promise.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.fromasync
+description: >
+  Array.fromAsync tries the various properties in order and awaits promises
+includes: [asyncHelpers.js, compareArray.js, temporalHelpers.js]
+flags: [async]
+features: [Array.fromAsync]
+---*/
+
+asyncTest(async function () {
+  const actual = [];
+  const items = {};
+  TemporalHelpers.observeProperty(actual, items, Symbol.asyncIterator, undefined, "items");
+  TemporalHelpers.observeProperty(actual, items, Symbol.iterator, undefined, "items");
+  TemporalHelpers.observeProperty(actual, items, "length", 2, "items");
+  TemporalHelpers.observeProperty(actual, items, 0, Promise.resolve(2), "items");
+  TemporalHelpers.observeProperty(actual, items, 1, Promise.resolve(1), "items");
```

Comment:
Well... perhaps this is a sign we need to separate out these observer APIs into a separate helper file; they certainly aren't specific to Temporal. But I'd say that's out of scope for this PR.

I think it could be simplified to this:
```js
const items = TemporalHelpers.propertyBagObserver(actual, {
  length: 2,
  0: Promise.resolve(2),
  1: Promise.resolve(1),
}, "items");
```
(and similar for some of the tests below with repeated calls to TemporalHelpers.observeProperty)

Reply:
This causes issues when comparing the output, because they're arrays of `TemporalHelpers.toPrimitiveObserver` objects.

Reply:
Fair enough.